src/zope/lifecycleevent/README.rst

```diff
@@ -0,0 +1,196 @@
+:class:`~.IZopeLifecycleEvent` interface) for easily sending that
+event in a single function call.
+
+.. note:: The convenience functions are simple wrappers for
+   constructing an event object and sending it via
+   :func:`zope.event.notify`. We will only demonstrate some examples
+   of this equivalence.
+
+.. note:: This document will not discuss actually *handling* these
+   events (setting up *subscribers* for them). For more information on
+   that topic, see `zope.event's documentation
+   <http://zopeevent.readthedocs.io/en/latest/classhandler.html>`_
+   (for basic uses) or :doc:`XXXWRITEME` for more flexible uses.
+
+We will go through the events in approximate order of how they would
+be used to follow the life-cycle of an object.
+
+
+Creation
+========
+
+The first event is :class:`.~IObjectCreatedEvent`, implemented by
+:class:`.~ObjectCreatedEvent`, which is used to communicate that a single object
+has been created. It can be sent with the
+:func:`zope.lifecycleevent.created` function.
+
+
+For example:
+
+    >>> from zope.event import notify
+    >>> from zope.lifecycleevent import ObjectCreatedEvent
+
+    >>> obj = {}
+    >>> notify(ObjectCreatedEvent(obj))
+
+Alternately, using the higher-level API, we could write that like
+this:
+
+    >>> from zope.lifecycleevent import created
+    >>> created(obj)
+
+Copying
+=======
+
+Copying an object is a special case of creating one. It can happen at
+any time and is implemented with :class:`~.IObjectCopiedEvent`,
+:class:`~.ObjectCopiedEvent`, or the API
+:func:`zope.lifecycleevent.copied`.
+
+    >>> import pickle
+    >>> copy = pickle.loads(pickle.dumps(obj))
+
+    >>> from zope.lifecycleevent import copied
+    >>> copied(copy, obj)
+
+Addition
+========
+
+After objects are created, it is common to *add* them somewhere for
+storage or access. This can be accomplished with the
+:class:`~IObjectAddedEvent` and its implementation
+:class:`~.ObjectAddedEvent`, or the API
+:func:`zope.lifecycleevent.added`.
+
+    >>> from zope.lifecycleevent import ObjectAddedEvent
+    >>> from zope.lifecycleevent import added
+
+    >>> container = {}
+    >>> container['name'] = obj
+    >>> added(obj, container, 'name')
+
+
+    >>> obj['modified'] = True
+
+
+Modification
+============
+
+One of the most common types of events used from this package is the
+:class:`IObjectModifiedEvent` (implemented by
+:class:`ObjectModifiedEvent`) that represents object modification.
+
+In the simplest case, it may be enough to simply notify interested
+parties that the object has changed. Like the other events, this can
+be done manually or through the convenience API
+(:func:`zope.lifecycleevent.modified`):
+
+    >>> obj['key'] = 42
+
+    >>> from zope.lifecycleevent import ObjectModifiedEvent
+    >>> notify(ObjectModifiedEvent(obj))
+
+The above is equivalent to this:
+
+    >>> from zope.lifecycleevent import modified
+    >>> modified(obj)
```

Comment:
You've been showing the high-level helpers in the examples above, so I think it would flow better if you showed that here first.  Then you can remind the reader that it's equivalent to the long-form ObjectModifiedEvent() and then segue into providing additional information via optional ObjectModifiedEvent() constructor arguments.